test/parallel/test-fs-watch-recursive.js

```diff
@@ -213,6 +213,26 @@ tmpdir.refresh();
   });
 })().then(common.mustCall());
 
+
+(async () => {
+  // Assert recursive watch does not leak handles
+  const rootDirectory = fs.mkdtempSync(testDir + path.sep);
+  const testDirectory = path.join(rootDirectory, 'test-7');
+  const filePath = path.join(testDirectory, 'only-file.txt');
+  fs.mkdirSync(testDirectory);
+
+  const watcher = fs.watch(testDirectory, { recursive: true });
+  watcher.on('change', common.mustCall(async () => {
+    await setTimeout(common.platformTimeout(100));
+    watcher.close();
```

Comment:
We need to only close the watcher if the event is triggered from the line 233. Can you follow the same code logic on this file and make sure that on exit, the watcher is closed.